### PR TITLE
Add typing to CI; Fix typing

### PR DESCRIPTION
**Pull request by MatthewFlamm**
#68 started adding type hints.  This PR enforces correct type hinting by using mypy.

In doing so, a few fixes were required.

### CODE REVIEW

File: .mypy.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[mypy]
+
+[mypy-metar.*]
+ignore_missing_imports = True
```
Comment (MatthewFlamm) [inline .mypy.ini]:
```suggestion
ignore_missing_imports = True

```

File: pynws/forecast.py [hunk 2]
```diff
@@ -45,22 +47,22 @@ def __init__(self, properties: dict[str, Any]):
 
             units = prop_value.get("uom")
             units = units.split(":")[-1] if units else None
-            details[prop_name] = time_values, units
+            self.details[prop_name] = time_values, units
```
Comment (MatthewFlamm) [inline pynws/forecast.py]: This line has a `str` for key and mypy would not accept `Detail` as a type.

File: pynws/forecast.py [hunk 3]
```diff
@@ -20,16 +20,18 @@
 
 ONE_HOUR = timedelta(hours=1)
 
+_T_Detail_str = Union[Detail, str]
```
Comment (MatthewFlamm) [inline pynws/forecast.py]: We should allow users to use strings if they want

**Comment by MatthewFlamm** [comment]
@lymanepp can you review this as it impacts #78 .  I'm not sure which would be better to merge first.

**Comment by MatthewFlamm** [comment]
#78 simplifies the return values in most cases, and introduces other typing.  Let's merge that first and then clean up here.

**Comment by lymanepp** [comment]
> #78 simplifies the return values in most cases, and introduces other typing. Let's merge that first and then clean up here.

I've been working on typing as well. We need to coordinate efforts on that. I agree that https://github.com/MatthewFlamm/pynws/pull/78 should be merged first to minimize impact.

**Review by lymanepp — APPROVED** [review]

**Pull request merged**